### PR TITLE
Replace: FIX: Normalize TSV end-of-line without stripping all whitespace

### LINKED ISSUE #1113: Validator should ensure TSV files do not contain a tab character before the linebreak
As discussed [here](https://github.com/bids-standard/bids-matlab/issues/96), the bids validator accepts this file from openneuro:
https://openneuro.org/crn/datasets/ds003195/snapshots/1.0.3/files/participants.tsv
that contains a tab at the end of each line. The question is whether this is indeed a compliant file.

### PR DESCRIPTION
**Pull request by sappelhoff**
fixes #1113 

replaces to #1114 ... cherry picking the relevant commits from there and putting them on an up to date master branch.

### CODE REVIEW

File: bids-validator/validators/tsv/__tests__/validateTsvColumns.spec.js [hunk 1]
```diff
@@ -55,4 +55,18 @@ describe('validateTsvColumns', () => {
...
+      {
+        contents: '\t',
+        file: file,
+      },
+      {
+        contents: '\t\t',
+        file: file,
+      },
+    ]
+    const issues = validateTsvColumns(tsvs, {})
```
Comment (effigies) [inline bids-validator/validators/tsv/__tests__/validateTsvColumns.spec.js]: Extra parameter added in ce5a154ff8d1365bf9d427abf5d31a0351544ecf.

```suggestion
    const issues = validateTsvColumns(tsvs, {}, {})
```

We should update the docs as well:

https://github.com/bids-standard/bids-validator/blob/d048d087a1ca4e7cac0310da275cd4a45e1c425a/bids-validator/validators/tsv/validateTsvColumns.js#L48-L53

**Comment by effigies** [comment]
Tried to install locally to track down the test failure, but can't get a Node environment that will run tests. Good luck!

**Comment by sappelhoff** [comment]
> Tried to install locally to track down the test failure, but can't get a Node environment that will run tests. Good luck!

I appreciate the effort :smile:

**Comment by rwblair** [comment]
Thanks @effigies that did it.

**Comment by rwblair** [comment]
I'm seeing this generate errors on ds000117 in bids-examples. I made it a global search replace in this commit e59124e

@effigies  Thoughts on if this is an overtly bad idea? I don't think people are sprinkling carriage returns in willy nilly.

**Comment by effigies** [comment]
Nope, looks right. I assumed replace was global.

**Pull request merged**